src/ln/functional_tests.rs

```diff
@@ -5352,3 +5352,261 @@ fn test_update_fulfill_htlc_bolt2_after_malformed_htlc_message_must_forward_upda
+
+	const VALUE_DUST: u64 = 10000;
+
+	let mut nodes = create_network(2);
+	let (_, _, channel_id, _) = create_announced_chan_between_nodes_with_value(&nodes, 0, 1, VALUE_DUST, 10001);
+	route_payment(&nodes[0], &[&nodes[1]], VALUE_DUST);
+
+	let node_a_output_value = {
+		let chan_lock = nodes[0].node.channel_state.lock().unwrap();
+		let last_local_commitment_txn = &chan_lock.by_id.get(&channel_id).unwrap().last_local_commitment_txn;
+		last_local_commitment_txn[0].output[0].value
+	};
+
+	let node_b_output_value = {
+		let chan_lock = nodes[1].node.channel_state.lock().unwrap();
+		let last_local_commitment_txn = &chan_lock.by_id.get(&channel_id).unwrap().last_local_commitment_txn;
+		last_local_commitment_txn[0].output[0].value
+	};
+
+	assert!(node_a_output_value < VALUE_DUST);
+	assert_eq!(node_a_output_value, node_b_output_value);
+
+	nodes[1].node.get_and_clear_pending_events();
+}
+
+#[test]
+fn test_commitment_signed_bolt2_sending_node_must_include_htlc_signature_corresponding_commitment_txn() {
+	//BOLT 2 Requirement: A receiving node:
+	//    * MUST include one htlc_signature for every HTLC transaction corresponding to the ordering of the commitment transaction (see BOLT #3).
+
+	let mut nodes = create_network(2);
+	let (chan_update_1, _, channel_id, _) = create_announced_chan_between_nodes(&nodes, 0, 1);
+
+	let route = nodes[0].router.get_route(&nodes[1].node.get_our_node_id(), None, &[], 1000000, TEST_FINAL_CLTV).unwrap();
+
+	let mut htlcs = vec![];
+
+	let session_priv = SecretKey::from_slice(&{
+		let mut session_key = [0; 32];
+		rng::fill_bytes(&mut session_key);
+		session_key
+	}).expect("RNG is bad!");
+
+	const NUM_HTLCS: u32 = 3;
+
+	for i in 0..NUM_HTLCS {
+		let cur_height = nodes[0].node.latest_block_height.load(Ordering::Acquire) as u32 + i;
+
+		let (_, payment_hash) = get_payment_preimage_hash!(nodes[0]);
+		let onion_keys = onion_utils::construct_onion_keys(&Secp256k1::signing_only(), &route, &session_priv).unwrap();
+		let (onion_payloads, _htlc_msat, htlc_cltv) = onion_utils::build_onion_payloads(&route, cur_height).unwrap();
+		let onion_packet = onion_utils::construct_onion_packet(onion_payloads, onion_keys, &payment_hash);
+
+		{
+			let mut chan_state_lock = nodes[0].node.channel_state.lock().unwrap();
+			let chan_state = chan_state_lock.borrow_parts();
+			let chan = chan_state.by_id.get_mut(&channel_id).unwrap();
+
+			let update_add_htlc = chan.send_htlc(1000000, payment_hash, htlc_cltv, HTLCSource::OutboundRoute {
+				route: route.clone(),
+				session_priv: session_priv.clone(),
+				first_hop_htlc_msat: 1000,
+			}, onion_packet).unwrap().unwrap();
+
+			htlcs.push(update_add_htlc.clone());
+		}
+	}
+
+	{
+		let mut chan_state_lock = nodes[0].node.channel_state.lock().unwrap();
+		let chan_state = chan_state_lock.borrow_parts();
+		let chan = chan_state.by_id.get_mut(&channel_id).unwrap();
+
+		let (commitment_signed, _) = chan.send_commitment().unwrap();
```

Comment:
Hmmm....I suppose this is testing *something*, but it may be interesting to check that they're sorted the same as the latest local commitment transaction, then try to deliver them out-of-order and make sure the receive side blows up. Though I'm somewhat skeptical we don't already test this pretty well.